File: chartiq/src/main/java/com/chartiq/sdk/DataSource.java
package com.chartiq.sdk;

import java.util.Map;

public interface DataSource {

    void pullInitialData(Map<String, Object> params, DataSourceCallback callback);

    void pullUpdateData(Map<String, Object> params, DataSourceCallback callback);

    void pullPaginationData(Map<String, Object> params, DataSourceCallback callback);
}

File: chartiq/src/main/java/com/chartiq/sdk/DataSource.kt
package com.chartiq.sdk

import com.chartiq.sdk.model.QuoteFeedParams

interface DataSource {

    fun pullInitialData(params: QuoteFeedParams, callback: DataSourceCallback)

    fun pullUpdateData(params: QuoteFeedParams, callback: DataSourceCallback)

    fun pullPaginationData(params: QuoteFeedParams, callback: DataSourceCallback)
}
